### DISCUSSION TITLE
Bash is waiting for me to press q in order to continue when running command, how do I stop it?

### DISCUSSION THREAD

**Discussion by eyalya** [ohmybash/oh-my-bash#391] [Q&A] [answered]
When I run command that has an output, for example aws cli command, I see the result but need to press q in order to continue. I think this is cause by using less. When I run script and just want to see the printing on the screen, and I dont want to press after every output. 
How do I cancel it (even temporary only for the command). 
Thanks

**Comment by akinomyoga** [comment]
> When I run command that has an output,

So does the problem happen even with e.g. `echo hello`? Usually, this kind of behavior is configured for each command, but do you mean you need to press `q` for *every command that has an output*?

> for example aws cli command,

`AWS_PAGER` is the environment variable that controls the behavior of `aws` CLI command, but I don't think oh-my-bash contains the settings for `AWS_PAGER`.

What is the result of the following command?

```bash
$ echo "$AWS_PAGER"
```

**Comment by eyalya** [comment]
Hi, 
The variable is empty. 
Before I installed the oh-my-bash it was running without it

**Reply by akinomyoga** [reply]
Thanks for checking the variable `AWS_PAGER`. So, it means that other settings are causing the problem. Does this problem happen with every command that has any output? For example, does it happen even with `echo hello`?

**Reply by akinomyoga** [reply]
Ah, OK. [The AWS document about cli pagination](https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-pagination.html) says that `aws` is affected also by the environment variable `PAGER`. In fact, oh-my-bash sets `PAGER` to `less` on [this line](https://github.com/ohmybash/oh-my-bash/blob/b9295c6da0dbec8d58ddb4a2b0edddff0d0b7530/lib/misc.sh#L6).

You can just unset the variable `PAGER` after sourcing `oh-my-bash.sh` in your `~/.bashrc`. You can add the following line at the end of `~/.bashrc`:

```bash
unset -v PAGER
```

If you would like to turn off the paging of AWS while keeping the environment variable `PAGER` for other commands, you may instead edit `~/.aws/config` to make `cli_pager` empty as described in [this secion](https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-pagination.html#cli-usage-pagination-clipager).

**Reply by akinomyoga** [reply]
Did this solve your problem? I have created an answer https://github.com/ohmybash/oh-my-bash/discussions/391#discussioncomment-4927494 for a brief conclusion.

**Comment by akinomyoga** [comment, marked as answer]
Here I summarize an answer for other users to easily find a solution.

Oh My Bash sets the environment variable `PAGER=less`. If you do not want it, you can unset it after `source "$OSH"/oh-my-bash.sh` in your `~/.bashrc`. For example, at the end of your `~/.bashrc`, you can add the following line:

```bash
# End of bashrc

unset -v PAGER
```

If you wouldn't like to use the pager at all instead of using the default specified in other places, you may instead try setting an empty value to `PAGER` (suggested by @eyalya):

```bash
# End of bashrc

export PAGER=
```

If you would like to turn it off for each command, you need to configure it for each command. For example, in the case of the `aws` CLI command, you can specify it through [`cli_pager` in `~/.aws/config`](https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-pagination.html#cli-usage-pagination-clipager).

**Reply by eyalya** [reply]
Hi, 
When I run in terminal `unset -v PAGER` it didn't remove it, but when I run `export PAGER=` it did work. I will leave it, but now I know how to cancel it when running scripts. Thanks

**Reply by akinomyoga** [reply]
Thanks for the information! Hmm, OK. I'll update the answer.

**Reply by akinomyoga** [reply]
I have updated it. Thank you!
